### ISSUE TITLE
BNF comments not supported 

### ISSUE THREAD

**Issue by mmoderau** [vallentin/vscode-bnf#2]
According to [specification](https://www.w3.org/Notation.html), a comment in a BNF file starts with a semicolon (`;`) and extends to the end of the line.

This extension supports only line comments starting with `#` and block comments between `(*` and `*)`. The latter is EBNF-style comment, the former is not copatible with neither BNF, EBNF or ABNF.

**Comment by vallentin** [comment]
Originally, I excluded `;` for comments as it is also used for termination in EBNF.
I guess, whether it is highlighted as a comment, doesn't really matter for EBNF, as text wouldn't be written after it. So I can change it to comments instead, if that is more desired?

**Comment by mmoderau** [comment]
Actually, EBNF allows multiple rules per line, separated by a semicolon.

EBNF and BNF syntaxes are not very compatible. I think, in the long run, it's better to create two separate languages in the extension.

**Comment by mmoderau** [comment]
After reading more about BNF, I guess there is not much of a specification, contrary to EBNF. Every source has its own variants and extensions and presents it as "the BNF".

As such, I guess support for BNF should not be the top priority.

I switched to EBNF for my project, anyway.

**Comment by vallentin** [comment]
> EBNF allows multiple rules per line, separated by a semicolon

Makes sense, but honestly never tried that, so I don't even know if the extension currently supports that.

All in all, I guess it would be best to just leave it be as is? Maybe in the future add an option to be able to toggle between BNF and EBNF, but that's beyond what I currently have time for.

I'll leave the issue open.
